src/map/clan.c

```diff
@@ -0,0 +1,839 @@
+		}
+	}
+	if (antagonists != NULL) {
+		const char *antg_const;
+
+		for (i = 0; i < libconfig->setting_length(antagonists) && i < clan->maxalliances; i++) {
+			if ((antg_const = libconfig->setting_get_string_elem(antagonists, i)) != NULL)
+				safestrncpy(c->antagonists[i].constant, antg_const, NAME_LENGTH);
+		}
+	}
+	clan->read_db_additional_fields(c, clan_, n, source);
+	idb_put(clan->db, c->clan_id, c);
+	c->tid = timer->add(timer->gettick() + c->kick_time, clan->inactivity_kick, c->clan_id, 0);
+
+	return true;
+}
+
+/**
+ * Reads Clan DB included by clan configuration file.
+ *
+ * @param settings The Settings Group from config file.
+ * @param source File name.
+ */
+void clan_read_db(struct config_setting_t *settings, const char *source)
+{
+	struct config_setting_t *clans;
+
+	nullpo_retv(settings);
+
+	if ((clans = libconfig->setting_get_member(settings, "clans")) != NULL ) {
+		int i;
+		int clan_count = libconfig->setting_length(clans);
+		for (i = 0; i < clan_count; i++) {
+			clan->read_db_sub(clans, i, source);
+		}
+		ShowStatus("Done reading '" CL_WHITE "%d" CL_RESET "' entries in '" CL_WHITE "%s" CL_RESET "'.\n", clan_count, source);
+		clan->constants();
+	}
+}
+
+/**
+ * Reads clan config file
+ *
+ * @param bool clear Whether to clear clan->db before reading clans
+ */
+bool clan_config_read(bool clear)
+{
+	struct config_t clan_conf;
+	struct config_setting_t *data = NULL;
+	const char *config_filename = "conf/clans.conf"; // FIXME: hardcoded name
+
+	if (clear) {
+		struct DBIterator *iter = db_iterator(clan->db);
+		struct clan *c_clear;
+
+		for (c_clear = dbi_first(iter); dbi_exists(iter); c_clear = dbi_next(iter)) {
+			if (c_clear->buff.script) {
+				script->free_code(c_clear->buff.script);
+			}
+			if (c_clear->tid != INVALID_TIMER) {
+				timer->delete(c_clear->tid, clan->inactivity_kick);
+			}
+		}
+		dbi_destroy(iter);
+		clan->db->clear(clan->db, NULL);
+	}
+
+	if (!libconfig->load_file(&clan_conf, config_filename)) {
+		return false;
+	}
+
+	data = libconfig->lookup(&clan_conf, "clan_configuration");
+
+	if (data != NULL) {
+		struct config_setting_t *settings = libconfig->setting_get_elem(data, 0);
+		int kicktime = 0, kickchecktime = 0;
+
+		libconfig->setting_lookup_int(settings, "MaxMember", &clan->max);
+		libconfig->setting_lookup_int(settings, "MaxAlliances", &clan->maxalliances);
+		libconfig->setting_lookup_int(settings, "InactivityKickTime", &kicktime);
+		if (!libconfig->setting_lookup_int(settings, "InactivityCheckTime", &kickchecktime)) {
+			ShowError("clan_config_read: failed to find InactivityCheckTime using official value.\n");
+			kickchecktime = 24;
+		}
+
+		// On config file we set the time in hours but here we use in seconds
+		clan->kicktime = 60 * 60 * kicktime;
+		clan->checktime = 60 * 60 * max(kickchecktime, 1) * 1000;
+
+		clan->config_read_additional_settings(settings, config_filename);
+		clan->read_db(settings, config_filename);
+	}
```

Comment:
In the event that the setting is not found, please `ShowError` here. (silent exits cause more confusion!!)

Reply:
Done inside `clan->read_db` function.